liesel/band.py

```diff
@@ -0,0 +1,232 @@
+from functools import partial
+
+import jax
+import jax.numpy as jnp
+import numpy as np
+import scipy.sparse as sps
+
+
+# @jax.jit not useful, f is compiled inside scan anyway, this makes it possible to return int.
+def bandwidth(full) -> int:
+    """
+    Returns the bandwidth p of `full` as an integer.
+
+    ## Parameters
+
+    - `full`: A full (n x n) matrix.
+    """
+
+    def f(j, row):  # Iterate over Row
+        def g(i, val):  # Iterate over value in Row
+            return i + 1, jax.lax.cond(val != 0, jnp.abs, lambda _: -1, i - j)
+
+        _, L = jax.lax.scan(g, 0, row)
+        return j + 1, L
+
+    _, b = jax.lax.scan(f, 0, full)
+    return int(jnp.max(b))
```

Comment:
We don't need to force an `int` here. An `int` wrapped in a `DeviceArray` should be fine

Reply:
Implemented.